skbio/io/util.py

```diff
@@ -42,7 +47,17 @@ def _get_filehandle(filepath_or, *args, **kwargs):
     pass through.
     """
     if _is_string_or_bytes(filepath_or):
-        fh, own_fh = open(filepath_or, *args, **kwargs), True
+        if requests.compat.urlparse(filepath_or).scheme in {'http', 'https'}:
+            sess = CacheControl(requests.Session(),
+                                cache=FileCache(gettempdir()))
+            req = sess.get(filepath_or, **kwargs)
+
+            # if the response is not 200, an exception will be raised
+            req.raise_for_status()
+
+            fh, own_fh = BytesIO(req.content), True
```

Comment:
Yes that is the case. IIRC it is already in memory i.e. the content
attribute from the request.

On (Apr-29-15|16:01), Evan Bolyen wrote:
> > @@ -42,7 +47,17 @@ def _get_filehandle(filepath_or, *args, **kwargs):
> >      pass through.
> >      """
> >      if _is_string_or_bytes(filepath_or):
> > -        fh, own_fh = open(filepath_or, *args, **kwargs), True
> > +        if requests.compat.urlparse(filepath_or).scheme in {'http', 'https'}:
> > +            sess = CacheControl(requests.Session(),
> > +                                cache=FileCache(gettempdir()))
> > +            req = sess.get(filepath_or, **kwargs)
> > +
> > +            # if the response is not 200, an exception will be raised
> > +            req.raise_for_status()
> > +
> > +            fh, own_fh = BytesIO(req.content), True
> 
> Does this load the request entirely into memory? 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/scikit-bio/pull/903/files#r29390201